dbms/src/Storages/KVStore/tests/gtest_region_persister.cpp

```diff
@@ -245,6 +248,220 @@ try
 }
 CATCH
 
+
+TEST_F(RegionSeriTest, FlexibleRestore)
+try
+{
+    auto ext_cnt_1 = 0;
+    auto ext_cnt_2 = 0;
+    auto ext_cnt_3 = 1;
+    auto ext_cnt_4 = 2;
+    {
+        auto counter = std::make_shared<int>(0);
+        // V3 store, V3 load, no unrecognized fields
+        auto region = std::make_shared<Region>(createRegionMeta(1001, 1));
+        region->updateRaftLogEagerIndex(5678);
+        FailPointHelper::enableFailPoint(FailPoints::force_region_persist_version, std::make_pair(3, ext_cnt_3));
```

Comment:
```suggestion
        // V2 store, V2 load, no unrecognized fields
        auto region = std::make_shared<Region>(createRegionMeta(1001, 1));
        region->updateRaftLogEagerIndex(5678);
        FailPointHelper::enableFailPoint(FailPoints::force_region_persist_version, std::make_pair(2, ext_cnt_3));
```

Reply:
The test to test if we can successfully save and load the extension fields when we have a V3 in the future

Reply:
But we need to ensure everything work well when reading with v2

Reply:
There is already a test about V4 -> V2. Do you mean we should have another test for V2 -> V2.